### DISCUSSION TITLE
Help - Serving Vaultwarden from homeserver to offshore server via haproxy

### DISCUSSION THREAD

**Discussion by ghost** [dani-garcia/vaultwarden#1930] [Q&A]
Hello I am seeking some advice from the community, I have asked around on Reddit and Matrix but I haven't had any luck.
As the title suggests I am trying to serve my Vaultwarden instance from my homeserver to my offshore server via haproxy so I can get the benefit of privacy and having a domain name (my home IP is dynamic). 

So using this method works for non-SSL connections, however as soon as I try using SSL, it won't connect. I am just really not sure where to proceed here, I have spent over 24 hours trying different things but no luck. 

Here is a little "diagram" of how I want it to run:
vault.example.com > 123.456.789.112 > HaProxy > Wireguard tunnel > Home Server > Vaultwarden.

**Comment by Masgalor** [comment]
> So using this method works for non-SSL connections, however as soon as I try using SSL, it won't connect.

Maybe you could elaborate on that a little more.

If the connection works without SSL my first guess would be that you are trying to serve a selfsigned certificate from your homeserver that is not accepted on your client. In that case you should get a real cert from a trusted CA and do the SSL termination on your HaProxy.
